### ISSUE TITLE
Dashboard views that may crash your browser should be gated

### ISSUE THREAD

**Issue by birdsarah** [dask/distributed#2979]
If I have a dask task graph with 200k tasks then clicking on the "tasks" tab of dask dasboard will lead to doom. Specifically it will just hang my browser. This is particularly disastrous if you're using dask-jupyterlab because then you break your jupyterlab session. 

I'm wondering if we can implement a gate that, for a given threshold (say 50k for arguments sake), delays the loading of the plot and presents the user a message saying "this view has more than 50k items, are you sure you want to display" or something like that.

**Comment by mrocklin** [comment]
Sounds good to me.

Or WebGL?

I think that we currently limit this to 100k in the tasks tab.  We could also reduce this limit (I think that it's configurable?)  I don't think that we do any kind of limit on the graph tab, which fails earlier at a few 10s of thousands.

**Comment by mrocklin** [comment]
For these plots we could also think about alternative visualizations that did a bit of aggregation.  (although that's probably a separate issue)

**Comment by birdsarah** [comment]
WebGL will alleviate this to a certain extent but it's not a total fix as the data still needs to arrive. 

Limiting seems less good as how else would you see the complete history.

That said, we could add a slider that defaults to nk history items, but then the user could increase that history.

I like the slider idea and is easy for tasks, but i think graph will need to be gated.

**Comment by mrocklin** [comment]
Yeah, gating graph makes sense regardless.  The graph loses a lot of value anyway with that many points.

**Comment by moseymosey** [comment]
We are seeing the same behavior with a graph containing 180k nodes. Once such a large task is complete, the scheduler keeps all the timing metrics in memory in case the UI attempts to display anything (which is the point at which the browser will freeze, the scheduler might crash due to mem limits, etc.).

Until there is some sort of fix for the UI, is it possible to programatically tell the scheduler to go ahead and drop successfully processed nodes from memory? After we complete large jobs such as these, we do not require them to hang around in the scheduler any longer. Tips or pointers on achieving this would be appreciated.

**Comment by mrocklin** [comment]
It's much much easier to just fix the UI.  If anyone wants to dive into the scheduler code here is an example place to start: 

https://github.com/dask/distributed/blob/eda27bee472090f5370121d7d35c201d2facd79e/distributed/dashboard/components/scheduler.py#L1111

One fix might be to modify the `update` method to pass if there are over a certain number of tasks.

**Comment by moseymosey** [comment]
There is still an issue though. When the scheduler gets very large graphs,it is holding on to the timing results of all those nodes. I have seen instances where a scheduler running in a containeer with a large amount of reserved memory will eventually blow the limit and get killed when new (potentially large) graphs get submitted. Normally, I believe it will drop older nodes in order to make room for the new (I am guessing, based on watching its behavior). But in some cases, it just fails. 

If a user application has no need to keep the timing info for all those processed nodes, and there have been no exceptions, would it make sense to have an API to clear them out of the scheduler? Or am I misunderstanding the intended usage here?

**Comment by mrocklin** [comment]
It sounds like you're describing a different issue.  I recommend raising a
separate issue.
